src/main/java/com/parpiiev/time/utils/dto/mappers/ActivityMapper.java

```diff
@@ -0,0 +1,33 @@
+package com.parpiiev.time.utils.dto.mappers;
+
+import com.parpiiev.time.model.Activity;
+import com.parpiiev.time.model.Category;
+import com.parpiiev.time.utils.dto.ActivityDTO;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ActivityMapper implements DtoMapper<ActivityDTO, Activity>{
+
+    @Override
+    public ActivityDTO mapToDto(Activity activity) {
+
+        ActivityDTO dto = new ActivityDTO();
+        dto.setActivity_id(activity.getId());
+        dto.setName(activity.getName());
+        dto.setActivity_category_id(activity.getCategory().getId());
+        return dto;
+    }
+
+    @Override
+    public Activity mapFromDto(ActivityDTO activityDTO) {
+        Activity activity = new Activity();
+
+        activity.setName(activityDTO.getName());
+
+        Category category = new Category();
+        category.setId(activityDTO.getActivity_category_id());
+        activity.setCategory(category);
+
+        return activity;
+    }
+}
```

Comment:
it's totally OK to use custom interfaces like this one, but you can use an existing and well-known `Function` interface and create two classes `ActivityDtoToActivity` and `ActivityToActivityDto` which can then be used in `stream().map(ActivityDtoToActivity::apply)`.
General rule - keep classes as focused on [Single responsibility](https://www.digitalocean.com/community/conceptual_articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design) as possible
```suggestion
@Component
public class ActivityDtoToActivity implements Function<ActivityDTO, Activity> {

    @Override
    public Activity apply(ActivityDTO activityDTO) {
        Activity activity = new Activity();

        activity.setName(activityDTO.getName());

        Category category = new Category();
        category.setId(activityDTO.getActivity_category_id());
        activity.setCategory(category);

        return activity;
    }
}
```